roles/gpu_operator_build_push_image/files/build-push-image.sh

```diff
@@ -0,0 +1,59 @@
+#! /bin/bash
+
+function help() {
+    printf -- "Options:
+        -h: Help
+        -n: Name of image
+        -t: Tag for image
+        -d: Path/Name of Dockerfile
+        -g: Git repo containing Dockerfile
+        -p: Path/Name of git repo Dockerfile
+        -q: Quay.io Org/Repo
+        -a: Authfile for quay.io
+    "
+}
+
+while getopts n:t:g:d:p:q:a:h flag
+do
+    case "${flag}" in
+        h) help
+           exit 0 ;;
+        n) name=${OPTARG};;
+        t) tag=${OPTARG};;
+        g) repo=${OPTARG};;
+        d) dockerfile=${OPTARG};;
+        p) path=${OPTARG};;
+        q) quay="quay.io/${OPTARG}";;
+        a) authfile=${OPTARG};;
+        *) exit 1 ;;
+    esac
+done
+
+if [[ -z $repo && -z $dockerfile ]] ; then
+    printf -- "Either a git repo (-g) or Dockerfile (-d) is required\n"
+    exit 1
+elif [[ -n $repo && -n $dockerfile ]] ; then
+    printf -- "Cannot have both -g and -d\n"
+    exit 1
+elif [[ -n $repo && -z $path ]] ; then
+    printf -- "Must supply path to Dockerfile within git repo (-p)\n"
+    exit 1
+elif [[ -z $name || -z $tag ]] ; then
+    printf -- "Image name (-n) and tag (-t) required\n"
+    exit 1
+fi
+
+if [[ -n $dockerfile ]] ; then
+    podman build -t $name:tag -f $dockerfile
+elif [[ -n $repo ]] ; then
+    git clone $repo
```

Comment:
I would suggest:
```
git clone $repo repo
cd repo
...
```
and change
```
 path="training_results_v0.7/NVIDIA/benchmarks/ssd/implementations/pytorch"
#to  
 path="/NVIDIA/benchmarks/ssd/implementations/pytorch"
```
(I think quay.io requires the initial `/`, like an absolute path, I don't know if it's a common convention or not)

Reply:
Note: also use shallow